src/Server/ServerHelpers.ts

```diff
@@ -256,7 +256,8 @@ export function prestigeHomeComputer(homeComp: Server): void {
 
   //Reset RAM usage calculation for all scripts
   homeComp.scripts.forEach(function (script) {
-    script.ramUsage = null;
+    // Don't need to invalidate the whole module, just the ram usage.
+    delete script.ramUsage;
```

Comment:
Here too

Reply:
> There are two separate "phases" here, determining the RAM cost and actually compiling.

These are separate and are not intertwined. You should be able to compile a script without requiring a ram calculation (as long as you are providing a ramcost override), and you should be able to calculate a ramcost without requiring a compilation (e.g. using the mem command). Maybe right now it's still doing a ram calc even if an override is provided, but this can be fixed.

However, anything that invalidates ram usage, other than prestiging, should invalidate both ram usage and the module itself. Prestiging is the exception because singularity script costs can change based on node.

> I liked the globalModuleSequenceNumber better. It was a lazier system; writing only had to change one thing, instead of doing an entire walk.

Hard disagree here. Running a script with an already-initialized module is a common codepath and should be as lean as possible. Module invalidation, on the other hand, is only performed on infrequent code paths - overwriting scripts or their imports. If you don't "do the walk" at invalidation time, then you basically still have to do it at script run time to check if any imports have had their modules updated.

> it's the correctness I'm concerned about more than the performance here.

Invalidating during the event that causes an invalidation is more correct than invalidating when running.

Reply:
> Invalidating during the event that causes an invalidation is more correct than invalidating when running.

Would that make `mem` work properly when you update a module that a script imports, but not the script itself, and want to verify the effect on RAM usage? That'd sure be nice.

Reply:
> Would that make mem work properly when you update a module that a script imports, but not the script itself, and want to verify the effect on RAM usage? That'd sure be nice.

Does that not already work as expected?

If this didn't work as expected already, it should with this implementation. Modifying a script Y that is imported by script X will immediately remove existing ram calculations from both X and Y. Then running `mem` for script X would be forced to perform a fresh recalculation using the updated content of its imported script Y.